src/npcmove.cpp

```diff
@@ -2769,45 +2793,69 @@ void npc::set_destination()
     add_msg( m_debug, "New goal: %s at %d,%d,%d", dest_type.c_str(), goal.x, goal.y, goal.z );
 }
 
-void npc::go_to_destination()
+void npc::go_investigate()
+{
+    if( !has_investigation_interest() ) {
+        add_msg( m_debug, "called npc::go_investigate with no target" );
+        move_pause();
+        return;
+    }
+
+    investigation_time -= 1;
+    update_path( investigation_target );
+    if( path.size() > 1 ) {
+        move_to_next();
+    } else {
+        investigation_time = 0;
+        investigation_target = no_goal_point;
+        move_pause();
+    }
+}
+
+void npc::go_to_goal()
 {
     if( goal == no_goal_point ) {
-        add_msg( m_debug, "npc::go_to_destination with no goal" );
+        add_msg( m_debug, "npc::go_to_goal with no goal" );
         move_pause();
-        reach_destination();
+        reach_destination(goal);
         return;
     }
 
+    go_to_destination(goal);
+}
+
+void npc::go_to_destination(tripoint &destination)
```

Comment:
The parameter didn't exist. Something that doesn't exist can't be modified. Meaning the ability to modify a parameter had to be introduced by your changes.

It would be better if you made the parameter `const` to avoid bugs. If you need to modify it inside the function but don't need the modified value returned, make a copy inside the function.